src/test/java/no/unit/nva/cristin/projects/FetchCristinProjectsTest.java

```diff
@@ -450,6 +472,62 @@ void handlerReturnsProjectsWrapperWithCorrectNextResultsAndPreviousResultsWhenLi
         assertEquals(expectedPrevious, actualPrevious);
     }
 
+    @Test
+    void handlerReturnsMatchingProjectsFromGrantIdSearchWhenSuppliedWithOnlyNumber() throws Exception {
+        cristinApiClientStub = spy(cristinApiClientStub);
+
+        doReturn(new HttpResponseStub(CRISTIN_QUERY_PROJECTS_RESPONSE_JSON_FILE))
+            .when(cristinApiClientStub).queryProjects(any(), eq(QUERY_USING_GRANT_ID));
+
+        doThrow(RuntimeException.class)
+            .when(cristinApiClientStub).queryProjects(any(), eq(QUERY_USING_TITLE));
+
+        InputStream input = requestWithQueryParameters(Map.of(QUERY, GRANT_ID_EXAMPLE));
+        handler.handleRequest(input, output, context);
+        GatewayResponse<ProjectsWrapper> gatewayResponse = GatewayResponse.fromOutputStream(output);
+
+        assertEquals(HttpURLConnection.HTTP_OK, gatewayResponse.getStatusCode());
+        assertEquals(MediaType.APPLICATION_JSON, gatewayResponse.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    void handlerReturnsProjectsFromTitleSearchWhenSuppliedWithQueryStringIncludingNumber() throws Exception {
+        cristinApiClientStub = spy(cristinApiClientStub);
+
+        doThrow(RuntimeException.class)
+            .when(cristinApiClientStub).queryProjects(any(), eq(QUERY_USING_GRANT_ID));
+
+        doReturn(new HttpResponseStub(CRISTIN_QUERY_PROJECTS_RESPONSE_JSON_FILE))
+            .when(cristinApiClientStub).queryProjects(any(), eq(QUERY_USING_TITLE));
+
+        InputStream input = requestWithQueryParameters(Map.of(QUERY, GRANT_ID_EXAMPLE + RANDOM_TITLE));
```

Comment:
here again if I change the nature of the input (to being a number) the test succeeds.